RIOT-OS/RIOT
# cpu/sam0_common: use generic exti_config

### Contribution description

Creating an `exti_config` array for a new MCU manually is tedious and error prone.
Luckiely all information is already availiable in the vendor files.

Credit for this discovery & method goes to @Sizurka

The file was generated with

```C

int main(void) {
        puts("static const int8_t exti_config[PORT_GROUPS][32] = {");

        for (unsigned port = 1; port < 5; ++port) {
                printf("#if PORT_GROUPS >= %d\n{\n", port);
                for (unsigned pin = 0; pin < 32; ++pin) {
                        printf("#ifdef PIN_P%c%02uA_EIC_EXTINT_NUM\n", '@' + port, pin);
                        printf("    PIN_P%c%02uA_EIC_EXTINT_NUM,\n", '@' + port, pin);
                        printf("#else\n    -1,\n#endif\n");
                }
                printf("},\n#endif\n\n");
        }

        puts("};");

        return 0;
}
```

No changes in generated code are expected, but this makes adding new members
of the sam0 CPU families much easier.








### Testing procedure

External interrupts should work as before, in fact the same binary should be generated.

If your board has a module connected that uses GPIO interrupts (e.g. the radio in `samr21-xpro`), they should still continue to work.
If your board has a button, you can try `tests/periph_gpio`.

e.g. to configure an interrupt for the falling flank of the user button (`PB31`) on `same54-xpro` run

```
init_int 1 31 0
```

Pressing the button should trigger a message on the screen

*tested on `samr21-xpro`,  `saml21-xpro` and `same54-xpro`*


### Issues/PRs references

For samd21 and samr21 updated vendor files are needed as those defines were added in a later version than the one currently used by RIOT.

~~#13313 for samd21~~
~~#13350 for saml21~~
~~#13311 for samr21~~
~~#13402 for samr30~~
